assets/js/modules/adsense/util/data-mock.js

```diff
@@ -219,3 +245,16 @@ export function getAdSenseMockResponse( args ) {
 
 	return data;
 }
+
+/**
+ * Generates mock response for AdSense reports.
+ *
+ * @since n.e.x.t
+ *
+ * @param {wp.data.registry} registry Registry with all available stores registered.
+ * @param {Object}           options  Report options.
+ */
+export function provideAdSenseMockReport( registry, options ) {
+	registry.dispatch( STORE_NAME ).receiveGetReport( getAdSenseMockResponse( options ), { options } );
+	registry.dispatch( STORE_NAME ).finishResolution( 'getReport', [ options ] );
```

Comment:
The Analytics provider (and other `provide*` utilities) don't mark resolvers as finished, so we should probably leave this out for consistency. It shouldn't have any effect to not do this since the report is only fetched if no report exists yet for the given options.
```suggestion
```

Reply:
Updated.